src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -182,12 +180,11 @@ def arm_exception_handler(ex, fault_type, summary, return_if_not_found=False):
         if status_code // 100 == 4:
             telemetry.set_user_fault()
         telemetry.set_exception(exception=ex, fault_type=fault_type, summary=summary)
-        raise CLIError("Http operation error occured while making ARM request: " + str(ex) + "\nSummary: {}".format(summary))
+        raise ClientRequestError("Http operation error occured while making ARM request: " + str(ex) + "\nSummary: {}".format(summary))
```

Comment:
Here it should not be 'ClientRequestError' but 'AzureResponseError'. If you want we can differentiate based on status codes too: https://github.com/Azure/azure-cli/blob/119856d4f5984e5a4ef0cdb5bc233c6f07da2720/src/azure-cli-core/azure/cli/core/azclierror.py#L171

Reply:
Also, this case is not always user fault. Its user fault only if the status code is 4xx